### DISCUSSION TITLE
let returns a value instead of nothing

### DISCUSSION THREAD

**Discussion by pepdiz** [nushell/nushell#6511] [Ideas]
when you use let to assign a value to a variable, the result value of the complete expression is nothing:

```
〉(let a = 3) | describe
nothing
```
so we cannot use its value :

```
〉let a = 3 | describe 
 Error: nu::parser::unexpected_keyword (link)

  × Let statement used in pipeline.
   ╭─[entry #22:1:1]
 1 │ let a = 3 | describe
   · ─┬─
   ·  ╰── let in pipeline
   ╰────
  help: Assigning '3' to 'a' does not produce a value to be piped. If the pipeline result is meant to be assigned to 'a', use 'let a = (3 | ...)'.
```

the error message suggest us to use a convoluted alternative:

```
〉let a = (3 | describe) 
〉$a
int
```

but this is not what I want,  what I really like and want is that  `let a = 3 | describe` return int and also assign 3 to variable 'a'  and this can be done if   let  returns the value passed to it.   

What I mean is semantics of let is  'assign to variable the result of evaluate the expression passed and return that expression value'.

```
> let a = 3 | describe
int
> $a
3
```
This is in consonance with functional programming where (almost) every expression is a function and every function returns a value.

**Comment by fdncred** [comment]
in nushell you'd do it this way.
```
let a = 3
$a | describe
```

**Comment by pepdiz** [comment]
that's ok, but I still find useful to return a value, this make let a function (or command in nushell parlance) rather than a syntax form.

**Comment by kubouch** [comment]
We disallow `let` in pipelines because it is not clear whether you mean `(let a = 3) | describe` or `let a = (3 | describe)`. Previously, it was allowed but it caused a lot of bugs and confusion so that's why it is the way it is currently.

**Comment by theAkito** [comment]
Variable assignment is a classic case of having a statement, not an expression, so I find the current behaviour very reasonable.
I also very much support the previous comment.
